R/write_results.R

```diff
@@ -28,7 +28,7 @@ write_results <- function(
   level %||% stop("Must provide 'level'")
   investorname %||% stop("Must provide 'investorname'")
```

Comment:
export technically works, just a quick content question. I noticed that on variable VaR_tech_company there is NaNs, positive and negative numbers, is that expected?

Reply:
positive and negative is definitely expected. NaN is something I will check and should probably be handled differently even if the calculation were using correct inputs.

Reply:
any updates on this? :)

Reply:
I found a number pof things that were contributing to issues, but some are still open. Firstly, the inner join on the same columns was problmeatic, because "year" kicked out required rows (my initial reason for using the left_join I guess). I solved this by filtering on the start year before the join, as all the financials are constant over time anyways. I also removed the production and EF columns from the masterdata before joining as they are not required here. I think that part should be fine now.

Reply:
I checked again, and it turns out there are still 44 rows with NaNs for VaR_tech_company and tech_company_value_change (always technology oil) as well as 539 rows with NA on production_shock_perc. Again, I guess you can assess better if that is a problem :)

Reply:
The latter is related to pivoting the empty shocks from the shock scenario longer for all companies. It's always NA. This will change with https://dev.azure.com/2DegreesInvesting/2DegreesInvesting/_workitems/edit/1618
The former are the companies with net_profit_margin = 0.
This will be handled appropriately with the follow up tasks [ADO 1677](https://dev.azure.com/2DegreesInvesting/2DegreesInvesting/_workitems/edit/1677) and [ADO 2180](https://dev.azure.com/2DegreesInvesting/2DegreesInvesting/_workitems/edit/2180)